dockyard/common/container/container.py

```diff
@@ -0,0 +1,162 @@
+from dockyard.common import utils, link
+
+import json
+from pecan.core import redirect
+
+class Container(object):
+    def __init__(self):
+        pass
+
+    def list(self, name_id=None, query_params=None):
+        if name_id:
+            url = ('/containers/%s/json' % (name_id))
+        else:
+            url = '/containers/json'
+
+        if query_params:
+            url = ('%s?%s' % (url, query_params))
```

Comment:
rather than updating the url here, pass query_params as parameter to dispatch_get_request method, it would be taken care in the method itself